### PR TITLE
Handle postgres arrays

### PR DESCRIPTION
**Pull request by jmcarp**
The current behavior for postgres.ARRAY sort of works. SQLAlchemy knows
that the `python_type` of `ARRAY` columns is `list`, so we convert them
to `List` fields. But we currently ignore the nested types:
`ARRAY(Integer)` should convert to `List(Int)`, and `ARRAY(String)`
should convert to `List(Str)`. This patch inspects the nested type of
the `ARRAY` to build the appropriate `List`.

### CODE REVIEW

File: marshmallow_sqlalchemy/convert.py [hunk 1]
```diff
@@ -28,44 +40,19 @@ class ModelConverter(object):
...
         postgresql.UUID: fields.UUID,
         postgresql.MACADDR: fields.String,
         postgresql.INET: fields.String,
+        postgresql.JSON: fields.Raw,
+        postgresql.JSONB: fields.Raw,
+        postgresql.HSTORE: fields.Raw,
+        postgresql.ARRAY: _postgres_array_factory,
 
         mysql.BIT: fields.Integer,
-        mysql.TINYINT: fields.Integer,
```
Comment (sloria) [inline marshmallow_sqlalchemy/convert.py:46]: Why were these removed?

Comment (jmcarp) [inline marshmallow_sqlalchemy/convert.py:46]: I probably should've put this into its own PR, but the mappings that I deleted aren't necessary. Because `mysql.TINYINT().python_type` is `int`, the mappings in `Schema.TYPE_MAPPING` already handle this case. I think mappings are only needed here for columns that either don't define `python_type` or that need special logic, like `postgres.UUID` or `postgres.ARRAY`.

Semi-documented in commit message at https://github.com/jmcarp/marshmallow-sqlalchemy/commit/67b2eb7cc94edc35d2ad4d74481c17cb7011c3d6.

Comment (sloria) [inline marshmallow_sqlalchemy/convert.py:46]: I see. Thanks for the clarification.
